app/src/org/commcare/dalvik/activities/EntitySelectActivity.java

```diff
@@ -279,12 +279,10 @@ public void onClick(View v) {
                 adapter.registerDataSetObserver(this.mListStateObserver);
             }
         }
-        //cts: disabling for non-demo purposes
-        //tts = new TextToSpeech(this, this);
         restoreLastQueryString(TAG + "-" + KEY_LAST_QUERY_STRING);
 
         if (!isUsingActionBar()) {
-            searchbox.setText(lastQueryString);
+            searchbox.setText(getLastQueryString());

```

Comment:
This will still trigger the search behavior even if it is null, right? I think we likely need to simply not set the text if the last query is null